torchsynth/synth.py

```diff
@@ -150,28 +176,29 @@ def test_step(self, batch, batch_idx):
         return T(0.0, device=self.device)
 
     @property
-    def hyperparameters(self) -> Dict[Tuple[str, str], Any]:
-        assert len(list(self.parameters())) == len(list(self.named_parameters()))
+    def hyperparameters(self) -> Dict[Tuple[str, str, str], Any]:
+        """
+        Returns a dictionary of curve and symmetry hyperparameter values keyed
+        on a tuple of the module name, parameter name, and hyperparameter name
+        """
         hparams = []
-        # I don't understand why this returns stuff in non-deterministic order :\
-        for name, parameter in self.named_parameters():
-            if not ModuleParameter.is_parameter_frozen(parameter):
-                hparams.append(((name, "curve"), parameter.parameter_range.curve))
-                hparams.append(
-                    ((name, "symmetric"), parameter.parameter_range.symmetric)
-                )
+        for param_key, parameter in self.get_parameters().items():
+            hparams.append(((*param_key, "curve"), parameter.parameter_range.curve))
+            hparams.append(
+                ((*param_key, "symmetric"), parameter.parameter_range.symmetric)
+            )
```

Comment:
Wait a second, the signature changed to `Tuple[str, str, str]`. Where's the triple?

Reply:
Maybe be more explicit here rather than tricky like:

```
for (module_name, parameter_name), parameter in self.get_parameters().items()
```

Reply:
We also just need the keys, not the full items()